### ISSUE TITLE
sitewide, forms 508-defect-2 [AXE-CORE, SEMANTIC MARKUP, COGNITION]: Email inputs MUST be paired with labels

### ISSUE THREAD

**Issue by jenstrickland** [department-of-veterans-affairs/va-forms-system-core#315]
# [508-defect-2](https://github.com/department-of-veterans-affairs/va.gov-team/blob/master/platform/accessibility/guidance/defect-severity-rubric.md#508-defect-2)

```diff
! Team impacted: VSA BAM2; Project: MDOT MVP
```

**Feedback framework**

- **❗️ Must** for if the feedback must be applied 
- **⚠️Should** if the feedback is best practice 
- **✔️ Consider** for suggestions/enhancements 

## Description

On the 1 of 3 Veteran information screen, there are text inputs for email addresses. At present, the label that says "Email address" is in a `<p>`, but must be in a `<label>` with a for attribute whose value matches the id of the input field. Additionally, h4 is not valid HTML as child of element label in this context.

## Point of Contact

**VFS Point of Contact:** Jennifer

## Acceptance Criteria

~~As a mouse user, I want to be able to select the label of an input to move focus into the input.~~

As a user with cognitive considerations, I expect to see a label and input pairing consistently styled as throughout the rest of the site, with the label just above the text/email/search input or to the right of a radio/checkbox input, so that I am clearly able to understand what entry is expected.

## Environment

* Operating System: all
* Browser: 
* Server destination: staging

## Steps to Recreate

1. Enter ~~https://staging.va.gov/hearing-aid-batteries-and-accessories/veteran-information/addresses~~ NEW URL: https://staging.va.gov/health-care/order-hearing-aid-batteries-and-accessories/ while logged in as user 1 in browser
2. Inspect the Email address code
3. Verify that the code appears as below in the "Current code"

### Current code
```diff
<div class="schemaform-field-template vads-u-margin-bottom--3 schemaform-first-field">
- <label id="root_email-label" class="schemaform-label" for="root_email">
    <h4>Email address</h4>
  </label>
  <p>We will send an order confirmation email with a tracking number to this email address.</p>
- <p class="vads-u-margin-bottom--1">Email address</p>
  <div class="schemaform-widget-wrapper">
-   <input type="email" id="root_email" name="root_email" maxlength="80" class="va-input-large" value="">
  </div>
  <div></div>
</div>
```

### Recommended code
```diff
<div class="schemaform-field-template vads-u-margin-bottom--3 schemaform-first-field">

  <h4>Email address</h4>

  <p>We will send an order confirmation email with a tracking number to this email address.</p>

+ <label id="root_email" class="schemaform-label" for="root_email">Email address</label>
+ <input id="root_email" type="email" name="root_email" maxlength="80" class="va-input-large"/>
</div>
```

## WCAG or Vendor Guidance (optional)

* [W3C Labeling Controls](https://www.w3.org/WAI/tutorials/forms/labels/)

* [MDN Basic Form Hints](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/forms/Basic_form_hints) - "Note that each `<input>` element has an id, and each `<label>` element has a for attribute, indicating the id of the associated `<input>`.

## Screenshots or Trace Logs

![Screen Shot 2020-06-04 at 5 12 54 PM](https://user-images.githubusercontent.com/57469/83811256-77449300-a687-11ea-9785-6840db1115d5.png)

**Comment by taylorkaren** [comment]
THE WORK THAT NEEDS TO BE DONE FOR THIS TICKET: Label for email address input that needs to be fixed.

**Comment by caw310** [comment]
@taylorkaren , Another item we identified as part of FLT Backlog.

**Comment by obliviga** [comment]
Hi @jenstrickland, I get a blank page when I try to access this page: https://staging.va.gov/health-care/order-hearing-aid-batteries-and-accessories/order-form-2346/introduction

Please advise, thank you!

![image.png](https://images.zenhubusercontent.com/6283eaffdea4c21364d67e5a/193fa648-0813-4819-aedf-08b6bb624226)

**Comment by obliviga** [comment]
Hi @bkjohnson, thanks for that info. When I go to that URL locally and have vets-api running, it errors out with these console errors:

![image.png](https://images.zenhubusercontent.com/6283eaffdea4c21364d67e5a/43492686-7154-4598-835f-50ba5f46690a)

Any idea how I can get past this?

**Comment by bkjohnson** [comment]
I can't help you with `vets-api`, but I was able to [use a local mock API](https://github.com/department-of-veterans-affairs/vets-website#running-a-mock-api-for-local-development) without any issues

**Comment by obliviga** [comment]
Hi @bkjohnson, I ran `yarn mock-api --responses src/applications/hca/api/mockapi.js` successfully, but the page is still not loading. Is that the mockapi you run for this page? Thanks.

**Comment by bkjohnson** [comment]
No, I used the common mock. However, I just tried it with the file you used and didn't have any problem loading the page. I don't know exactly what you're seeing, but maybe you didn't set up localStorage to pretend like you're logged in.

**Comment by obliviga** [comment]
@bkjohnson Gotcha, I had to set the `hasSession` property to `true` in local storage.

**Comment by taylorkaren** [comment]
PR: https://github.com/department-of-veterans-affairs/vets-website/pull/21677

**Comment by obliviga** [comment]
PR has been merged: http://69b7713ef1205ab3937003a999b7685c.review.vetsgov-internal/health-care/order-hearing-aid-batteries-and-accessories/
